Source/santactl/Commands/sync/SNTCommandSyncManager.m

```diff
@@ -146,22 +145,21 @@ - (void)postEventsToSyncServer:(NSArray<SNTStoredEvent *> *)events isFromBundle:
   }
 }
 
-- (void)postBundleEventToSyncServer:(SNTStoredEvent *)event
-                              reply:(void (^)(SNTBundleEventAction))reply {
+- (void)postBundleEventToSyncServer:(SNTStoredEvent *)event reply:(void (^)(BOOL))reply {
+  if (!event) return;
   SNTCommandSyncState *syncState = [self createSyncState];
   SNTCommandSyncEventUpload *p = [[SNTCommandSyncEventUpload alloc] initWithState:syncState];
-  if (event && [p uploadEvents:@[event]]) {
+  if ([p uploadEvents:@[event]]) {
     if ([syncState.bundleBinaryRequests containsObject:event.fileBundleHash]) {
-      reply(SNTBundleEventActionSendEvents);
+      reply(YES); // store and immediately send related bundle events
       LOGD(@"Needs related events");
     } else {
-      reply(SNTBundleEventActionDropEvents);
       LOGD(@"Bundle event upload complete");
```

Comment:
Now this case doesn't reply back to santad. I am pretty sure santad will keep around the block and captured objects forever. It may also prevent subsequent calls to `postBundleEventToSyncServer:reply:` from being sent since XPC operates serially. Testing now...

Reply:
Objects that are captured within a reply block that is never executed do "leak".

Client:
```objc
  NSXPCConnection *c =
      [[NSXPCConnection alloc] initWithServiceName:@"com.google.corp.XPCTestService"];
  c.remoteObjectInterface =
      [NSXPCInterface interfaceWithProtocol:@protocol(XPCTestServiceProtocol)];
  [c resume];

  NSArray *hudge = [self arrayWithNumberOfStrings:1000000];

  [[c remoteObjectProxy] sortArray:[self arrayWithNumberOfStrings:42] withReply:^(NSArray *a) {
    NSLog(@"hudge %lu", hudge.count);
    NSLog(@"%lu", a.count);
  }];
```

Service:
```objc
- (void)sortArray:(NSArray *)array withReply:(void (^)(NSArray *))reply {
  NSLog(@"%@ called with count %lu", NSStringFromSelector(_cmd), array.count);
  if (array.count == 42) {
    reply([array sortedArrayUsingSelector:@selector(localizedCaseInsensitiveCompare:)]);
  }
}
```

The first run is with a reply block that fires, the second is one that does not fire.

<img width="1176" alt="screen shot 2017-09-13 at 4 23 32 pm" src="https://user-images.githubusercontent.com/2117646/30398872-f828acde-989f-11e7-9366-d1377b89c44f.png">

To my second point. Since the method you are calling does return, it does not block subsequent calls.

In summery we still need the enum for the reply.

Reply:
Oh well, I guess it was too good to be true

Reply:
I like the enum approach, it is explicit. Which is nice for when we re-read this code 1 year from now 😉